### ISSUE TITLE
Using the Razor inject directive

### ISSUE THREAD

**Issue by jerriep** [statiqdev/Statiq.Framework#89]
I was hopeful that I could use the Razor `@inject` directive to inject services that I registered with the Statiq bootstrapper into my Razor pages. Unfortunately, when I execute my pipeline, I get the following error message:

```
Error while executing module RenderRazor in PagesPipeline/PostProcess: No service for type 'xxx.Common.Data.Dao.IStoresDao' has been registered.
```

Looking at the source code, it appears that inside the `RazorCompiler` constructor you build up a separate list of services which is required for the Razor rendering pipeline. 

Do you think that it would be possible to extend the current Razor implementation to somehow customize these services?

My thinking is that it would be great if one can do something along the following lines:

```cs
new RenderRazor()
    .WithServices(serviceCollection => { ... }))
```

I am not sure whether there are technical issue preventing this, but if not, I am willing to create a PR for this with your guidance as to how you see this working.

**Comment by daveaglick** [comment]
That's absolutely something I'd expect to work. The separate service collection in the Razor module is a holdover from Wyam when there wasn't a central container. I've got a note somewhere to revisit that but you beat me to it :).

Let me see if I can have the Razor module support the main service collection for the next release - it lines up nicely with some other work I'm doing on module/library initialization.

**Comment by daveaglick** [comment]
Got this working now, just need to do a little cleanup. Look for a new release later tonight, it's been long enough anyway so a version bump is in order regardless.

The way it works under the hood is that there's a new `IInitializer` interface that any library can implement. The `Bootstrapper` (but _only_ if using the `Bootstrapper`) will find those implementations and run them automatically. The Statiq.Razor library contains an initializer implementation that will get automatically run when using the `Bootstrapper` and it registers all the extra Razor services into the primary service collection. Then when the `RenderRazor` module is executed, it checks if the required Razor services have been registered with the primary container and if so, it uses it. If not it creates a secondary container as before to register the Razor services in as a fallback and uses that.

TL;DR: as long as you're using the `Bootstrapper` your Razor views will have access to any services registered with the primary container:

```
public static class Program
{
    public static async Task<int> Main(string[] args) =>
        await Bootstrapper.Factory
            .ConfigureServices(x => x.AddSingleton<IMyService, MyServiceImpl>())
            .RunAsync();
}
```
```
@inject IMyService MyService
```

**Comment by jerriep** [comment]
Great stuff. Thank you @daveaglick

**Comment by daveaglick** [comment]
👍 alpha.29 just released

**Comment by jerriep** [comment]
Thanks again Dave 👍
